TransformationSystem/scripts/dirac-production-runjoblocal.py

```diff
@@ -0,0 +1,164 @@
+_jobID = None
+
+for switch in Script.getUnprocessedSwitches():
+  if switch [ 0 ] in ( 'D', 'Download' ):
+    _downloadinputdata = True
+    _jobID = switch[1]
+  if switch [ 0 ] in ( 'I', 'Protocol' ):
+    _downloadinputdata = False
+    _jobID = switch[1]
+
+def __runSystemDefaults(jobID, vo):
+  """
+  Creates the environment for running the job and returns
+  the path for the other functions.
+  
+  """
+
+  
+  tempdir = str(vo) + "job" + str(jobID) + "temp"
+
+  try:
+    os.mkdir(tempdir)
+    if not sys.exc_info()[1][0]:
+      S_OK("Temporary directory created.")    
+    pass
+  except:    
+    if sys.exc_info()[1][0] == 17:
+      S_OK("Temporary directory already exists.")
+    elif sys.exc_info()[1][0] == 30:
+      print sys.exc_info()[1], "Unable to create temporary directory"
+#      DError(errno.EROFS, "Unable to create temporary directory")
+
+    
+  basepath = os.getcwd()
+  return basepath + "/" + tempdir
+
+def __downloadJobDescriptionXML(jobID, basepath):
+  """
+  Downloads the jobDescription.xml file into the temporary directory
+  created.
+  
+  """
+  from DIRAC.Interfaces.API.Dirac import Dirac
+  jdXML = Dirac()
+  jdXML.getInputSandbox(jobID, basepath)
+
+def __modifyJobDescription(jobID, basepath, downloadinputdata):
+  """
+  Modifies the jobDescription.xml to, instead of DownloadInputData, it 
+  uses InputDataByProtocol
+  
+  """
+  if not downloadinputdata:
+    from xml.etree import ElementTree as et
+    archive = et.parse(basepath + "/InputSandbox" + str(jobID) + "/jobDescription.xml")
+    for element in archive.getiterator():
+      if element.text == "DIRAC.WorkloadManagementSystem.Client.DownloadInputData":
+        element.text = "DIRAC.WorkloadManagementSystem.Client.InputDataByProtocol"
+        archive.write(basepath + "/InputSandbox" + str(jobID) + "/jobDescription.xml")
+        S_OK("Job parameter changed from DownloadInputData to InputDataByProtocol.")
+
+  
+def __downloadPilotScripts(basepath, diracpath):
+  """
+  Downloads the scripts necessary to configure the pilot
+  
+  """
+  shutil.copyfile(str(diracpath) + "/WorkloadManagementSystem/PilotAgent/dirac-pilot.py"   , basepath + "/dirac-pilot.py")
+  shutil.copyfile(str(diracpath) + "/WorkloadManagementSystem/PilotAgent/pilotCommands.py" , basepath + "/pilotCommands.py")
+  shutil.copyfile(str(diracpath) + "/WorkloadManagementSystem/PilotAgent/pilotTools.py"    , basepath + "/pilotTools.py")
+      
+def __configurePilot(basepath, vo):
+  """
+  Configures the pilot.
+  This method was created specifically for LHCb pilots, more info
+  about othe VOs is needed to make it more general.
+  """
+
+  from DIRAC.ConfigurationSystem.Client.Helpers.CSGlobals    import getVO, getSetup
+  from DIRAC.ConfigurationSystem.Client.ConfigurationData    import gConfigurationData
+  
+  dir = str(os.getcwd())
+  os.rename(dir + '/.dirac.cfg', dir + '/.dirac.cfg.old')
+  os.system("cp " + dir + "/pilot.cfg " + dir + "/.dirac.cfg")
```

Comment:
Yes, in LHCb module I did not import shutil than I forgot about the copyfile, but in this case I cannot move the pilot.cfg because it is used on the setInputSandbox and setConfigArgs on _runJobLocally...